lldb/test/API/CMakeLists.txt

```diff
@@ -41,7 +41,7 @@ if(LLDB_TEST_SWIFT)
   set(LLDB_SWIFT_LIBS ${SWIFT_LIBRARY_DIR}/swift CACHE STRING "Path to swift libraries")
   # Prefer the just-built stdlib over the system one.
   set(SWIFT_TEST_ARGS
-    --inferior-env "DYLD_LIBRARY_PATH=\\\"${LLDB_SWIFT_LIBS}/macosx\\\""
+    --inferior-env "DYLD_LIBRARY_PATH=\\\"${LLDB_SWIFT_LIBS}/macosx:${LLVM_LIBRARY_OUTPUT_INTDIR}/lldb/clang/lib/darwin\\\""
     --inferior-env "LD_LIBRARY_PATH=\\\"${LLDB_SWIFT_LIBS}/${CMAKE_SYSTEM_PROCESSOR}\\\""
```

Comment:
@JDevlieghere @bulbazord 

This really surprises me. Why does our CMake file hardcode macOS-specific test args? I guess the answer is because the DYLD_* options don't have any effect on Linux? Still this seems wrong...